### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by claws-dev**
What changed?
- New test page UI
- All scraping info is fetched from TMDB instead of manually entered
- All providers now use the `year` correctly and accurately
- All TV provider now check for the season and episode more accurately
- SeriesFree now keeps track of it's promises and returns results before the `done` event fires
- Cleaned up a few provider's logic
- Defaulted the `DISABLE_PM2_LOGGING` to `true`
- 123Movies should work now
- Removed AfdahTV (afdah.to) until it's re-done for movies.

### CODE REVIEW

File: src/scrapers/providers/tv/SwatchSeries.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ const resolve = require('../../resolvers/resolve');
 
 async function SwatchSeries(req, sse) {
     const clientIp = req.client.remoteAddress.replace('::ffff:', '').replace('::1', '')
-    const showTitle = req.query.title;
-    const {season, episode} = req.query;
+    const showTitle = req.query.name;
```
Comment (ddivad) [inline src/scrapers/providers/tv/SwatchSeries.js]: Is this going to be changed in Kamino too? @claws-dev

File: src/scrapers/providers/universal/5movies.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const resolve = require('../../resolvers/resolve');
 
 async function _5movies(req, sse){
     logger.debug("5movies NOTICE: This primitively removes the year from the end of the title. If this source breaks, did you change the title parameter?");
-    let showTitle = req.query.title;
+    let showTitle = req.query.title || req.query.name;
```
Comment (SpajicM) [inline src/scrapers/providers/universal/5movies.js]: Why this?

File: src/scrapers/providers/tv/SwatchSeries.js [hunk 3]
```diff
@@ -93,7 +108,7 @@ async function SwatchSeries(req, sse) {
             });
         } catch (err) {
             if (!sse.stopExecution) {
-                logger.error({source: 'SwatchSeries', sourceUrl: url, query: {title: req.query.title, season: req.query.season, episode: req.query.episode}, error: (err.message || err.toString()).substring(0, 100) + '...'});
+                logger.error({source: 'SwatchSeries', sourceUrl: url, query: {title: req.query.name, season: req.query.season, episode: req.query.episode}, error: (err.message || err.toString()).substring(0, 100) + '...'});
```
Comment (SpajicM) [inline src/scrapers/providers/tv/SwatchSeries.js]: Why?

File: src/scrapers/providers/tv/Series8.js [hunk 4]
```diff
@@ -22,52 +23,58 @@ module.exports = class Series8 extends BaseProvider {
...
+                    }
                 }
-            });
-            if (!seasonLink) {
-                this.logger.debug('Series8', `Could not find: ${showTitle} Season ${season}`);
-                return Promise.all(resolvePromises);
+            }
+            if (!seasonLinkElement.length) {
+                // No season link.
+                logger.debug('Series8', `Could not find: ${showTitle} (${year}) Season ${season}`);
```
Comment (ddivad) [inline src/scrapers/providers/tv/Series8.js]: Should be `this.logger`

File: src/scrapers/providers/tv/SeriesFree.js [hunk 5]
```diff
@@ -92,33 +117,15 @@ async function SeriesFree(req, sse) {
...
+            videoUrls.forEach((videoUrl) => {
+                resolvePromises.push(scrapeHarder(videoUrl, userAgent, clientIp, sse, jar));
             });
         } catch (err) {
             if (!sse.stopExecution) {
               logger.error({
                 source: 'SeriesFree',
                 sourceUrl: url,
-                query: {title: req.query.title, season: req.query.season, episode: req.query.episode},
+                query: {title: req.query.name, season: req.query.season, episode: req.query.episode},
```
Comment (SpajicM) [inline src/scrapers/providers/tv/SeriesFree.js]: same

File: src/scrapers/providers/tv/AfdahTV.js [hunk 6]
```diff
@@ -9,11 +12,11 @@ const {padTvNumber} = require('../../../utils');
 
 async function AfdahTV(req, sse) {
     const clientIp = req.client.remoteAddress.replace('::ffff:', '').replace('::1', '');
-    const title = req.query.title;
+    const title = req.query.name;
```
Comment (SpajicM) [inline src/scrapers/providers/tv/AfdahTV.js]: same

File: src/scrapers/providers/tv/AfdahTV.js [hunk 7]
```diff
@@ -154,7 +157,7 @@ async function AfdahTV(req, sse) {
             }
         } catch (err) {
             if (!sse.stopExecution) {
-                logger.error({source: 'AfdahTV', sourceUrl: url, query: {title: req.query.title, season: req.query.season, episode: req.query.episode}, error: (err.message || err.toString()).substring(0, 100) + '...'});
+                logger.error({source: 'AfdahTV', sourceUrl: url, query: {title: req.query.name, season: req.query.season, episode: req.query.episode}, error: (err.message || err.toString()).substring(0, 100) + '...'});
```
Comment (SpajicM) [inline src/scrapers/providers/tv/AfdahTV.js]: same

File: src/scrapers/providers/tv/SwatchSeries.js (hunk 1 above)
Comment (claws-dev) [reply src/scrapers/providers/tv/SwatchSeries.js]: Nope, it was a mistake. I missed this one.

File: src/scrapers/providers/universal/5movies.js (hunk 2 above)
Comment (claws-dev) [reply src/scrapers/providers/universal/5movies.js]: Ignore 5movies, it's not used.

File: src/scrapers/providers/tv/AfdahTV.js (hunk 6 above)
Comment (claws-dev) [reply src/scrapers/providers/tv/AfdahTV.js]: Ignore AfdahTV, it's not used.

File: src/scrapers/providers/tv/AfdahTV.js (hunk 7 above)
Comment (claws-dev) [reply src/scrapers/providers/tv/AfdahTV.js]: Ignore AfdahTV, it's not used.

File: src/scrapers/providers/tv/SeriesFree.js (hunk 5 above)
Comment (claws-dev) [reply src/scrapers/providers/tv/SeriesFree.js]: Gotcha

File: src/scrapers/providers/tv/SwatchSeries.js (hunk 3 above)
Comment (claws-dev) [reply src/scrapers/providers/tv/SwatchSeries.js]: Gotcha.

**Comment by SpajicM** [comment]
Getting this error. And when this happens, no done event is sent.
> {"message":{"source":"SwatchSeries","providerUrl":"http://gorillavid.in/ns45rurz25dm","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://daclips.in/coi7r45zsrgy","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://movpod.in/zg9szfv2ej7h","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://movpod.in/5dsvii0hyd10","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://daclips.in/32ncfxqhnmzo","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
(node:12840) UnhandledPromiseRejectionWarning: RequestError: Error: ESOCKETTIMEDOUT
    at new RequestError (C:\Users\RedJohn\Desktop\Claws\node_modules\request-promise-core\lib\errors.js:14:15)
    at Request.plumbing.callback (C:\Users\RedJohn\Desktop\Claws\node_modules\request-promise-core\lib\plumbing.js:87:29)
    at Request.RP$callback [as _callback] (C:\Users\RedJohn\Desktop\Claws\node_modules\request-promise-core\lib\plumbing.js:46:31)
    at self.callback (C:\Users\RedJohn\Desktop\Claws\node_modules\request\request.js:185:22)
    at Request.emit (events.js:182:13)
    at ClientRequest.<anonymous> (C:\Users\RedJohn\Desktop\Claws\node_modules\request\request.js:819:16)
    at Object.onceWrapper (events.js:273:13)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.emitRequestTimeout (_http_client.js:661:40)
    at Object.onceWrapper (events.js:273:13)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket.Socket._onTimeout (net.js:449:8)
    at ontimeout (timers.js:425:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
    at Timer.processTimers (timers.js:212:10)
From previous event:
    at Request.plumbing.init (C:\Users\RedJohn\Desktop\Claws\node_modules\request-promise-core\lib\plumbing.js:36:28)
    at Request.RP$initInterceptor [as init] (C:\Users\RedJohn\Desktop\Claws\node_modules\request-promise-core\configure\request2.js:41:27)
    at new Request (C:\Users\RedJohn\Desktop\Claws\node_modules\request\request.js:127:8)
    at request (C:\Users\RedJohn\Desktop\Claws\node_modules\request\index.js:53:10)
    at SeriesFree.RequestPromise.defaults.target (C:\Users\RedJohn\Desktop\Claws\src\scrapers\providers\tv\SeriesFree.js:26:16)
    at C:\Users\RedJohn\Desktop\Claws\node_modules\request\index.js:100:12
    at scrapeHarder (C:\Users\RedJohn\Desktop\Claws\src\scrapers\providers\tv\SeriesFree.js:30:37)
    at videoUrls.forEach (C:\Users\RedJohn\Desktop\Claws\src\scrapers\providers\tv\SeriesFree.js:121:38)
    at Array.forEach (<anonymous>)
    at scrape (C:\Users\RedJohn\Desktop\Claws\src\scrapers\providers\tv\SeriesFree.js:120:23)
(node:12840) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:12840) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
{"message":{"source":"SwatchSeries","providerUrl":"http://daclips.in/t46xrtkszfc9","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://gorillavid.in/4xx28sp2nmnv","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":"resolving: http://thevideo.me/4dj4wa5b6pdt","level":"debug"}
{"message":{"source":"SwatchSeries","providerUrl":"http://gorillavid.in/c4kban1gm9nc","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":"resolving: http://www.speedvid.net/a3anpfqhtocs","level":"debug"}
{"message":"resolving: http://thevideo.me/4v2ufh3056b7","level":"debug"}
{"message":"resolving: http://www.speedvid.net/bpbmqoktg5ul","level":"debug"}
{"message":"resolving: http://thevideo.me/oasykg2quwfk","level":"debug"}
{"message":{"source":"SwatchSeries","providerUrl":"http://daclips.in/l4olu4w1yks4","error":"Error: ETIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://speedvid.net/embed-wmkhuiv2ec0m.html","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://speedvid.net/embed-7dhzymwxugbi.html","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}
{"message":{"source":"SwatchSeries","providerUrl":"http://speedvid.net/embed-3n7gbivtmz2e.html","error":"Error: ESOCKETTIMEDOUT..."},"level":"error"}

**Comment by claws-dev** [comment]
I added a try/catch to the block of code that's causing the above error.

File: src/scrapers/providers/index.js [hunk 8]
```diff
@@ -2,23 +2,23 @@
 
 module.exports = exports = {
     movies: [
-        require('./movies/Afdah'),
-        require('./movies/AZMovies'),
-        require('./movies/bfmovies'),
-        require('./movies/StreamM4u'),
-        require('./movies/MovieFiles'),
+        // require('./movies/Afdah'),
```
Comment (ddivad) [inline src/scrapers/providers/index.js]: Think you forgot to un-comment these from this file? @claws-dev

Comment (claws-dev) [reply src/scrapers/providers/index.js]: 🤦‍♂️

**Pull request merged**